packages/grafana-data/src/utils/matchers/matchers.ts

```diff
@@ -0,0 +1,40 @@
+import { Field, DataFrame } from '../../types/dataFrame';
+import { Registry, RegistryItemWithOptions } from '../registry';
+
+export type DataMatcher = (series: DataFrame, field?: Field) => boolean;
```

Comment:
@ryan going through the code. Seems like I get the feeling what what we are about to achieve with this concept.

One thing thad bugs me here is what is your thinking behind pushing series as an argument of a matcher?

Reply:
Perhaps we should have a FieldMatcher and a FrameMatchet?